lib/puppet/pops/lookup/hiera_config.rb

```diff
@@ -106,7 +106,18 @@ def self.config_exist?(config_root)
   def self.symkeys_to_string(struct)
     case(struct)
     when Hash
-      Hash[struct.map { |k,v| [k.to_s, symkeys_to_string(v)] }]
+      map = struct.map { |k,v|
+        case k.class.to_s
+        when "Symbol"
+          key = k.to_s
+        else
+          key = k
+        end
+        value = symkeys_to_string(v)
+        [key, value]
+      }
+
+      Hash[map]
```

Comment:
The algorithm used is a slow way of mapping a hash (the original was too). It is faster to do as follows:
```
map = {}
struct.each {|k,v| map[ k.is_a?(Symbol) ? k.to_s : k] = symkeys_to_string(v) }
map
```
And - for the future note that a `case` with a value can match with its type against a class - in the PR it takes the class name as a String and then compares against `"Symbol"` instead of just matching the value against `Symbol`. 

We have found in benchmarks that it is faster to build up a hash key by key, than first creating an array of array hash entries and then creating a hash from that.